### PR TITLE
Revert LED Matrix Interleaving

### PR DESCRIPTION
**Pull request by WDobert**

### CODE REVIEW

**Review by alanssitis — APPROVED** [review]
LGTM at a glance. Merge when it's been tested.

File: firmware-nucleo/smart-air-hockey-table/App/driver_led.c [hunk 1]
```diff
@@ -62,37 +65,49 @@ void Driver_LED_SetColor(uint_fast8_t col, uint_fast8_t row, Color color)
...
+		LL_DMA_EnableChannel(GPDMA1, LL_DMA_CHANNEL_2);
+		LL_DMA_EnableChannel(GPDMA1, LL_DMA_CHANNEL_3);
 	}
 }
 
-void GPDMA1_Channel0_Handler()
+void GPDMA1_Channel0123_Handler()
 {
-	is_transfer_active = false;
+	transfers_active--;
```
Comment (alanssitis) [inline firmware-nucleo/smart-air-hockey-table/App/driver_led.c:112]: Is the handler called every time a channel is flushed?

Comment (alanssitis) [reply firmware-nucleo/smart-air-hockey-table/App/driver_led.c:112]: Are the calls to the handler atomic? Cuz if not we have a data race that will result in errors.

File: firmware-nucleo/smart-air-hockey-table/App/driver_led.c [hunk 2]
```diff
@@ -62,37 +65,49 @@ void Driver_LED_SetColor(uint_fast8_t col, uint_fast8_t row, Color color)
...
 	is_transfer_requested = true;
 }
 
 void Driver_LED_Clear()
 {
-	for (uint_fast32_t i = 0; i < LED_DATA_LENGTH; i++)
+	for (uint_fast32_t channel = 0; channel < LED_CHANNELS; channel++)
 	{
-		led_buffer[i] = COLOR_COMPARE_OFF;
+		for (uint_fast32_t i = 0; i < LED_DATA_LENGTH / LED_CHANNELS; i++)
```
[the remark this answers is not in the document]
Comment (WDobert) [reply firmware-nucleo/smart-air-hockey-table/App/driver_led.c:79]: They aren't equivalent values, `LED_CHANNEL_LENGTH` includes the reset period, but we don't want to write to that.

[the remark this answers is not in the document]
Comment (alanssitis) [reply firmware-nucleo/smart-air-hockey-table/App/driver_led.c:79]: I see, lemme make another comment.

File: firmware-nucleo/smart-air-hockey-table/App/driver_led.c [hunk 3]
```diff
@@ -9,25 +9,28 @@
 #define LED_DATA_LENGTH (LED_MATRIX_PIXEL_COUNT * LED_COLOR_DATA_SIZE)
 #define LED_RESET_LENGTH (LED_CHANNELS * LED_RESET_MSG_SIZE)
 #define LED_BUFFER_LENGTH (LED_DATA_LENGTH + LED_RESET_LENGTH)
+#define LED_CHANNEL_LENGTH (LED_BUFFER_LENGTH / LED_CHANNELS)
```
Comment (alanssitis) [inline firmware-nucleo/smart-air-hockey-table/App/driver_led.c:12]:
```suggestion
#define LED_DATA_LENGTH (LED_MATRIX_PIXEL_COUNT * LED_COLOR_DATA_SIZE)
#define LED_CHANNEL_DATA_SIZE (LED_DATA_LENGTH / LED_CHANNELS)
#define LED_CHANNEL_LENGTH (LED_DATA_LENGTH + LED_RESET_MSG_SIZE)

```

File: firmware-nucleo/smart-air-hockey-table/App/driver_led.c (hunk 2 above)
Comment (alanssitis) [inline firmware-nucleo/smart-air-hockey-table/App/driver_led.c:79]:
```suggestion
		for (uint_fast32_t i = 0; i < LED_CHANNEL_DATA_LENGTH; i++)
```

File: firmware-nucleo/smart-air-hockey-table/App/driver_led.c (hunk 3 above)
Comment (alanssitis) [reply firmware-nucleo/smart-air-hockey-table/App/driver_led.c:12]: do something like the above

**Review by alanssitis — CHANGES_REQUESTED** [review]

**Review by alanssitis — APPROVED** [review]
lgtm

**Pull request merged**